# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/builtin/TestingFunctions.cpp
# Commit: 9a162cc369e3
# Full Hash: 9a162cc369e3b6f51f5767924e63d46539c4f631
# Author: Julien Pages <jpages@mozilla.com>
# Date: 2025-02-22 09:14:45
# Regressor Bug: 1861078
# File Overlap Count: 1
# Description:
#   Bug 1861078 - wasm: make metadata offsets relative to CodeBlock->codeBase. r=rhunt
#   
#   Metadata used offsets to the beginning of code segment. Change that to use
#   offsets relative to the beginning of CodeBlock. This allows us to avoid offseting
#   metadata later on and prepare the work for simplification of stackmaps.
# ==============================================================================

diff -r e543a77786c9 -r 9a162cc369e3 js/src/builtin/TestingFunctions.cpp
--- a/js/src/builtin/TestingFunctions.cpp	Fri Feb 21 17:09:51 2025 +0000
+++ b/js/src/builtin/TestingFunctions.cpp	Fri Feb 21 17:17:58 2025 +0000
@@ -1780,12 +1780,11 @@
     js::wasm::Instance& inst = fun->wasmInstance();
     const uint32_t funcIndex = fun->wasmFuncIndex();
     const js::wasm::CodeBlock& codeBlock = inst.code().funcCodeBlock(funcIndex);
-    const js::wasm::CodeSegment& segment = *codeBlock.segment;
     const js::wasm::FuncExport& func = codeBlock.lookupFuncExport(funcIndex);
     const js::wasm::CodeRange& codeRange = codeBlock.codeRange(func);
 
-    jit_begin = segment.base() + codeRange.begin();
-    jit_end = segment.base() + codeRange.end();
+    jit_begin = codeBlock.base() + codeRange.begin();
+    jit_end = codeBlock.base() + codeRange.end();
   } else if (fun->hasJitScript() && fun->nonLazyScript()->hasIonScript()) {
     sprinter.printf("; backend=ion\n");
     jit_begin = fun->nonLazyScript()->ionScript()->method()->raw();